infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesPersonalAccessTokenManager.java

```diff
@@ -354,28 +365,27 @@ public void forceRefreshPersonalAccessToken(String scmServerUrl)
         scmPersonalAccessTokenFetcher.refreshPersonalAccessToken(subject, scmServerUrl);
     gitCredentialManager.createOrReplace(personalAccessToken);
     store(personalAccessToken);
-    removePreviousTokensIfPresent(subject, scmServerUrl);
+    removePreviousTokenSecretsIfPresent(scmServerUrl);
   }
 
-  private void removePreviousTokensIfPresent(Subject subject, String scmServerUrl)
+  private void removePreviousTokenSecretsIfPresent(String scmServerUrl)
       throws ScmConfigurationPersistenceException, UnsatisfiedScmPreconditionException {
-    List<PersonalAccessToken> personalAccessTokens =
-        doGetPersonalAccessTokens(subject, null, scmServerUrl);
-    for (int i = 1; i < personalAccessTokens.size(); i++) {
-      PersonalAccessToken token = personalAccessTokens.get(i);
-      if (token.getScmProviderUrl().equals(scmServerUrl)) {
-        try {
-          String namespace = getFirstNamespace();
-          cheServerKubernetesClientFactory
-              .create()
-              .secrets()
-              .inNamespace(namespace)
-              .withName(token.getScmTokenId())
-              .delete();
-        } catch (InfrastructureException e) {
-          throw new ScmConfigurationPersistenceException(e.getMessage(), e);
+    try {
+      for (KubernetesNamespaceMeta namespaceMeta : namespaceFactory.list()) {
+        List<Secret> secrets = doGetPersonalAccessTokenSecrets(namespaceMeta);
+        for (int i = 1; i < secrets.size(); i++) {
```

Comment:
Should we remove all secrets or keep the last one?

Reply:
We need to keep the last one

Reply:
Sorry, I mixed up, according to https://github.com/eclipse-che/che-server/pull/719/files#diff-c0fa2851d5bdc5b65a29d8ac5d7d71d4a961b03f526f43c50027e82b7948f775R257 we need to keep the first one - the newest.